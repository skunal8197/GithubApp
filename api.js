export function APIRequest(who) {
  if (who === 'followers') {
    return fetch('https://api.github.com/users/skunal8197/followers').then(res => res.json())
  }

  if (who === 'following') {
    return fetch('https://api.github.com/users/skunal8197/following').then(res => res.json())
  }

  if (who === 'repos') {
    return fetch('https://api.github.com/users/skunal8197/repos').then(res => res.json())
  }

  if (who === 'profile') {
    return fetch('https://api.github.com/users/skunal8197').then(res => res.json())
  }

  else {
    return 'no argument provided'
  }
}
