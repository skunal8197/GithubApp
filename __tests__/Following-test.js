import 'react-native';
import React from 'react';
import App from '../App';
import FollowingScreen from '../screens/FollowingScreen';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { APIRequest } from '../api.js'

describe('Following snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  })
  it('renders correctly', () => {
    const tree = renderer.create(
      <FollowingScreen />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('calls the following endpoint and asserts the data passed in', () => {
    fetch.mockResponseOnce(JSON.stringify({ data: 'testData' }))
    APIRequest('following').then(res => {
      expect(res.data).toEqual('testData')
    })
    expect(fetch.mock.calls.length).toEqual(2)
    expect(fetch.mock.calls[0][0]).toEqual('https://api.github.com/users/skunal8197/following')
  })
});
