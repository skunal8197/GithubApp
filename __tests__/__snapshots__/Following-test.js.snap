// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers snapshot renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;

exports[`Following snapshot renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#bbb",
        "borderTopWidth": 1,
        "marginTop": 20,
      },
      undefined,
    ]
  }
/>
`;
